src/navigation_controller.ts

```diff
@@ -148,33 +159,65 @@ export class NavigationController {
   }
 
   /**
-   * Sets whether the navigation controller has focus. This will enable keyboard
-   * navigation if focus is now gained. Additionally, the cursor may be reset if
-   * it hasn't already been positioned in the workspace.
+   * Sets whether the navigation controller has toolbox or flyout focus. This
+   * will enable keyboard navigation in the toolbox or flyout.
    *
-   * @param workspace the workspace that now has input focus.
+   * @param workspace the workspace that now has toolbox/flyout input focus.
    * @param isFocused whether the environment has browser focus.
    */
-  setHasFocus(workspace: WorkspaceSvg, isFocused: boolean) {
-    this.hasNavigationFocus = isFocused;
+  updateToolboxOrFlyoutFocus(workspace: WorkspaceSvg, isFocused: boolean) {
     if (isFocused) {
-      this.navigation.focusWorkspace(workspace, true);
+      if (!workspace.getToolbox()) {
```

Comment:
nit: I would flip this to
```
if (workspace.getToolbox()) {
  this.navigation.focusToolbox(workspace);
} else {
  this.navigation.focusFlyout(workspace);
}
```

Fortunately none of the workspace we're testing with have neither a flyout nor a toolbox.

Reply:
And see comment in `src/index.ts` about potentially just making this handle the toolbox case and skipping the flyout case if you haven't tested it.

Reply:
Flipped the logic at first, but then changed it a bit more due to the other comment and removing explicit flyout support.